parlai/agents/hugging_face/dialogpt.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.agents.hugging_face.dict import DialoGPTDictionaryAgent
+from parlai.agents.hugging_face.gpt2 import Gpt2Agent, GPT2Decoder, HFGPT2Model
+from parlai.utils.misc import warn_once
+
+try:
+    from transformers import GPT2Model
+except ImportError:
+    raise ImportError('Please run `pip install transformers`.')
+
+
+############################################
+## Modules
+############################################
+
+
+class DialoGPTDecoder(GPT2Decoder):
+    """
+    DialoGPT Decoder.
+
+    This decoder is initialized with the pretrained model from Hugging Face.
+    """
+
+    def _init_from_pretrained(self, opt):
+        # load model
+        model_sz = opt['gpt2_size']
+        fle_key = f'microsoft/DialoGPT-{model_sz}'
+        return GPT2Model.from_pretrained(fle_key)
+
+
+class DialoGPTModel(HFGPT2Model):
+    """
+    Hugging Face DialoGPT Model.
+    """
+
+    def _get_decoder(self, opt, dict):
+        return DialoGPTDecoder(opt, dict)
+
+
+############################################
+## Agent
+############################################
+
+
+class DialogptAgent(Gpt2Agent):
+    """
+    Hugging Face DialoGPT Agent.
+
+    GPT2 is a multi-layer decoder-only Transformer.
+    The decoder is initialized with pretrained weights from Hugging Face.
+    Read more about this model here
+    <https://huggingface.co/transformers/model_doc/dialogpt.html>.
+
+    DialoGPT comes in three  sizes: small, medium, large.
+
+    If you are finetuning the Gpt2 agent as a dialogue agent, be sure
+    to run `--add-special-tokens True`. To examine the performance of the
```

Comment:
perhaps to enforce this, we can override `train_step` to raise an error if this opt value is `False`? or even just set an attribute in `__init__` so we don't have to access opt every step

Reply:
good idea

Reply:
there is actually a check in the init of the parent class, if bs > 1